pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -93,26 +123,38 @@ func (gce *GCECloud) AttachDisk(diskName string, nodeName types.NodeName, readOn
 	if err != nil {
 		return fmt.Errorf("error getting instance %q", instanceName)
 	}
-	disk, err := gce.getDiskByName(diskName, instance.Zone)
-	if err != nil {
-		return err
+
+	// Try fetching as regional PD
+	var disk *GCEDisk
+	if gce.AlphaFeatureGate.Enabled(GCEDiskAlphaFeatureGate) {
+		disk, err = gce.getRegionalDiskByName(diskName)
+		if err != nil {
+			glog.Warningf("Could not find regional PD named %q to Attach. Will look for a zonal PD", diskName)
```

Comment:
Give this a higher log level so that it doesn't spam the logs.

Reply:
If the caller has enough information to differentiate, we should probably consider passing it in.  (in a later pr)